app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,90 @@
+class TasksController < ApplicationController
+  TASKS = [
+    { name: "cleaning", description: "vacuum" },
+    { name: "dinner", description: "make soup" },
+    { name: "shopping", description: "Go to Target" },
+  ]
+
+  def index
+    @tasks = Task.all.order(:name)
+  end
+
+  def show
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to root_path
+    end
+  end
+
+  def new
+    @task = Task.new(name: "Default task")
+  end
+
+  def create
+    # puts params
+    task = Task.new(
+      name: params["task"]["name"],
+      description: params["task"]["description"],
+    )
```

Comment:
This is a place where you can refactor this to use `task_params` instead of outlining `name` and `description`